File: api/app/v1/models/order_state.go
package models

type OrderState struct {
	ID    uint            `json:"id,omitempty" gorm:"primary_key"`
	Name  string          `json:"name"`
	Value OrderStateValue `json:"value" gorm:"type:text"`
}

type OrderStateValue string

const (
	CreatedOrderStateValue    = "created"
	StartedOrderStateValue    = "started"
	CorteOrderStateValue      = "corte"
	ConfeccionOrderStateValue = "confeccion"
	CalidadOrderStateValue    = "calidad"
	EmpaqueOrderStateValue    = "empaque"
	FinishedOrderStateValue   = "finished"
	CanceledOrderStateValue   = "canceled"
)

func DefaultOrderStatus() []*OrderState {
	return []*OrderState{
		{Name: "Creada", Value: CreatedOrderStateValue},
		{Name: "Iniciada", Value: StartedOrderStateValue},
		{Name: "Corte", Value: CorteOrderStateValue},
		{Name: "Confección", Value: ConfeccionOrderStateValue},
		{Name: "Calidad", Value: CalidadOrderStateValue},
		{Name: "Empaque", Value: EmpaqueOrderStateValue},
		{Name: "Finalizada", Value: FinishedOrderStateValue},
		{Name: "Cancelada", Value: CanceledOrderStateValue},
	}
}

File: client/src/schema/orderState.ts
import type { MergeCoreCollection } from '../index.js';

/**
 * vytex_order_status type
 */
export type VytexOrderState<Schema = any> = MergeCoreCollection<
	Schema,
	'vytex_orders-status',
	{
		id: number;
		name: string;
		value: OrderStateValue;
	}
>;

export type OrderStateValue =
	| 'created'
	| 'started'
	| 'finished'
	| 'canceled'
	| 'corte'
	| 'confeccion'
	| 'calidad'
	| 'empaque';
